source/addonStore/models/addon.py

```diff
@@ -161,13 +163,21 @@ def isPendingInstall(self) -> bool:
 			# have not been installed yet
 			addonDataManager._downloadsPendingInstall,
 		)
+		nameInDownloadsPendingCompletion = filter(
+			lambda m: m.model.name == self.name,
+			# add-ons which are currently being downloaded
+			# and have not been cancelled
+			addonDataManager._downloadsPendingCompletion,
+		)
 		return (
 			super().isPendingInstall
 			# True if this add-on has been downloaded but
 			# has not been installed yet
 			or bool(next(nameInDownloadsPendingInstall, False))
 			# True if this add-on is currently being downloaded
+			# and the download has not been cancelled
 			or os.path.exists(self.tempDownloadPath)
+			and bool(next(nameInDownloadsPendingCompletion, False))
```

Comment:
Can you explain why this change is needed?
Shouldn't `os.path.exists` be false if the download has been cancelled?

Reply:
> Can you explain why this change is needed? Shouldn't `os.path.exists` be false if the download has been cancelled?

This is because the cleanup file is done in `_down`, which may not be deleted in time because of multithreading.